packages/commonwealth/client/scripts/views/components/Knock/Knock.tsx

```diff
@@ -0,0 +1,51 @@
+import {
+  KnockFeedProvider,
+  KnockProvider,
+  NotificationFeedPopover,
+  NotificationIconButton,
+} from '@knocklabs/react';
+import React, { useRef, useState } from 'react';
+import app from 'state';
+
+import '@knocklabs/react-notification-feed/dist/index.css';
+import './Knock.css';
+
+const KNOCK_PUBLIC_API_KEY =
+  'pk_test_Hd4ZpzlVcz9bqepJQoo9BvZHokgEqvj4T79fPdKqpYM';
+
+const KNOCK_IN_APP_FEED_ID = 'fc6e68e5-b7b9-49c1-8fab-6dd7e3510ffb';
+
+function useUserId(): number | undefined {
```

Comment:
we try to avoid adding custom hooks or util functions in the component. If this is global hook/util, we place it in hooks/utils directory in root. If it's "local", we create `utils.ts` file or `useCustomHook.ts` file next to the component where it is used. Also, please use `const = () => {}` syntax instead of the function - for project consistency 

Having said all that, I don't believe we need it whatsoever. It is not reactive and it is based on app state object. We usually use the app.X reference in the code (which is not ideal, but at least we would be consistent)

Reply:
Thanks. Long story but I was using this pattern because I had to call /status during dev but I backed it out now..